lisa/tests/kernel/scheduler/load_tracking.py

```diff
@@ -203,7 +203,9 @@ def _test_task_signal(self, signal_name, allowed_error_pct,
                           trace, cpu, task_name, capacity=None):
         exp_util = self.get_expected_util_avg(trace, cpu, task_name, capacity)
         signal_df = self.get_task_sched_signals(trace, cpu, task_name, [signal_name])
-        signal_mean = signal_df[UTIL_AVG_CONVERGENCE_TIME_S:][signal_name].describe()['mean']
+        signal = signal_df[UTIL_AVG_CONVERGENCE_TIME_S:][signal_name]
+
+        signal_mean = area_under_curve(signal) / (signal.index[-1] - signal.index[0])
```

Comment:
Indexes are monotonic (https://github.com/ARM-software/trappy/commit/7bb0bd9f29711e2a40e88fa33ae04fcb7df94e56), and I have no idea if Pandas/Numpy records this for optimizing min/max. So directly using an index seems like the better choice to me.

Reply:
Yes, we had a sanitization step somewhere in trappy which grants that... I just find the code above more "robust" and (maybe) a bit more easy to read... although I've also personally always used your same style. That was just to possibly consider to change our style... but not sure if the "robustness" argument is "robust" enough ;)

Reply:
I prefer using pandas.Series.mean() function. I find the `area_under_curve()/duration` usage convoluted. Besides the behavior to re-sample and interpolate before it does the integration is semi magic. Most importantly around decay time where there's a big gap of no samples until the task wakes up again.

If it does fix a problem then we need to understand exactly why. Can you add a few debug messages to help with that? I suggest calculate using the 2 methods and print them out in a log.debug() or something.

By the way, why you had `___.describe()['mean']` instead of `___.mean()` in the removed line? Curious what's the difference.

Reply:
> I prefer using pandas.Series.mean() function. I find the `area_under_curve()/duration` usage convoluted. Besides the behavior to re-sample and interpolate before it does the integration is semi magic. Most importantly around decay time where there's a big gap of no samples until the task wakes up again.
> 

That was my feeling as well, so I went for `mean()` until I hit issues and reconsidered (see below).

> If it does fix a problem then we need to understand exactly why. Can you add a few debug messages to help with that? I suggest calculate using the 2 methods and print them out in a log.debug() or something.
> 

I did that and the difference is noticeable. With the integral mean I can safely set a 5% threshold (usual error % is < 2%), with the regular mean I need at least 10%. I had a chat with @derkling and the integration *is* the way to go - IIRC that can be verified on the PELT simulator. He'll 100% explain it better than I, but that integration is required.

> By the way, why you had `___.describe()['mean']` instead of `___.mean()` in the removed line? Curious what's the difference.

There's none, that wasn't necessary :)